### PR TITLE
Feature user application

### PR DESCRIPTION
**Pull request by Burito2021**
Revised taking into account the feedback from Maxim

### CODE REVIEW

File: src/main/java/net/library/controller/UserController.java [hunk 1]
```diff
@@ -25,14 +27,23 @@ public UserController(final UserService service) {
...
     }
 
-    @GetMapping()
-    public ResponseEntity<List<User>> getUsers() {
+    @PostMapping()
+    public ResponseEntity<LibraryResponse> addUser(@Valid @RequestBody UserRequest userRequest) {
+
+         service.addUser(userRequest);
+        return ResponseEntity.ok(LibraryResponse.of(MdcUtils.getCid()));
+    }
```
Comment (MaximDzhezhelo) [inline src/main/java/net/library/controller/UserController.java:41]: Add logic
- retrieve by id
- delete by id

File: src/main/java/net/library/converter/UserConverter.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package net.library.converter;
+
+import net.library.model.entity.User;
+import net.library.model.request.UserRequest;
+
+public class UserConverter {
+
+    public static User of(final UserRequest userRequest){
```
Comment (MaximDzhezhelo) [inline src/main/java/net/library/converter/UserConverter.java:8]: Move this method into the Request

File: src/main/java/net/library/model/response/UserResponse.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.List;
+
+public class UserResponse {
+
+    @JsonProperty("x-Correlation-id")
+    private String xCorrelationId;
+
+    @JsonProperty("users")
+    private List<User> users;
```
Comment (MaximDzhezhelo) [inline src/main/java/net/library/model/response/UserResponse.java:15]: you shouldn't expose the inner structure and return the whole object. Use Dto and return only special fields. these can be all fields or truncated ones

File: src/main/java/net/library/model/response/UserResponse.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import net.library.model.entity.User;
+
+import java.util.List;
+
+public class UserResponse {
+
+    @JsonProperty("x-Correlation-id")
+    private String xCorrelationId;
```
Comment (MaximDzhezhelo) [inline src/main/java/net/library/model/response/UserResponse.java:12]: This field doesn't belong to this object. Could you move it in a general wrapper?

File: src/main/java/net/library/service/UserService.java [hunk 5]
```diff
@@ -1,39 +1,37 @@
...
+        return StreamSupport.stream(userRepository.findAll().spliterator(), false).toList();
     }
 
-    public void deleteAllUsers(){
+    public void addUser(final UserRequest userRequest) {
 
-        users.clear();
-        LOGGER.info("All the users deleted");
+        final var newUser = UserConverter.of(userRequest);
+        userRepository.save(newUser);
```
Comment (MaximDzhezhelo) [inline src/main/java/net/library/service/UserService.java:31]: what if this user already exists? 
I don't see this logic, moreover I don't see even tests for this business logic

File: src/test/resources/application-test.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  sql:
+    init:
+      mode: always
+      schema-locations: 'classpath:sql/schema.sql'
+      data-locations: 'classpath:sql/data.sql'
+
+  test:
+    database:
+      replace: none
+logging:
```
Comment (MaximDzhezhelo) [inline src/test/resources/application-test.yml:20]: use logging configuration in one place. For this purpose, you have logback file

File: src/test/java/net/library/controller/ControllerTest.java [hunk 7]
```diff
@@ -1,16 +1,21 @@
...
 import org.springframework.test.web.servlet.MockMvc;
 import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
 
-import static net.library.util.HttpUtil.BASE_URI;
+import static net.library.util.HttpUtil.GLOBAL_BASE_URI;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
+@ActiveProfiles("test")
+@ExtendWith(SpringExtension.class)
```
Comment (MaximDzhezhelo) [inline src/test/java/net/library/controller/ControllerTest.java:18]: Why do you need to use it?

File: src/main/java/net/library/model/repository/UserRepository.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package net.library.model.repository;
```
Comment (MaximDzhezhelo) [inline src/main/java/net/library/model/repository/UserRepository.java:1]: repository level does not belong to the model. move it higher

**Review by MaximDzhezhelo — CHANGES_REQUESTED** [review]

File: src/test/java/net/library/controller/ControllerTest.java (hunk 7 above)
Comment (Burito2021) [reply src/test/java/net/library/controller/ControllerTest.java:18]: @ActiveProfiles("test")  or @ExtendWith(SpringExtension.class)?

**Pull request closed without merging**